### DISCUSSION TITLE
Disclosing private notes and other unnecessary fields in Sends API response?

### DISCUSSION THREAD

**Discussion by MagnusKryger** [dani-garcia/vaultwarden#1490] [General]
Hi,

I'm not a 100% this is working as intended hence i put this under discussion - but it seems very unlogic to me why it would be as it would be a potential security risk i believe.
Also, in the official bitwarden service, information like this is not disclosed.

The issue I found, is that when accessing a Send from it's public link, when taking a look at the API response stuff like private notes is disclosed (even though it's not plaintext, I don't think it's supposed to be publicly available???), along with other fields i don't think should be in there for the public view aswell. I quickly highlighted some of them I feel is very unnecessary, but some might be a requirement idk.
These fields is nowhere found looking at the same API response when using the official Bitwarden service.
![2021-03-15_23-55-29](https://user-images.githubusercontent.com/26845215/111231692-fccb7b00-85e9-11eb-9056-83726dc43d62.png)

From the vaults perspective
![2021-03-16_00-04-42](https://user-images.githubusercontent.com/26845215/111232573-a3fce200-85eb-11eb-95e4-d1816737aaba.png)

Setup (latest testing tag as of writing this)
### Your environment (Generated via diagnostics page)
* Bitwarden_rs version: v1.19.0-a71359f6
* Web-vault version: v2.19.0

**Comment by dani-garcia** [comment]
Yeah, out of all those fields, only the following seem required:
```
Id
Type
Name
File
Text
ExpirationDate
Object
```

Which means we can remove quite a bit of fields from that output. I'll change that tomorrow.

**Reply by dani-garcia** [reply]
Fixed in https://github.com/dani-garcia/bitwarden_rs/commit/84810f2bb22a161b7ef5b6e081acde5cadcfd155
